report/report.go

```diff
@@ -58,18 +58,49 @@ type Report struct {
 // MakeReport makes a clean report, ready to Merge() other reports into.
 func MakeReport() Report {
 	return Report{
-		Endpoint:       NewTopology(),
-		Address:        NewTopology(),
-		Process:        NewTopology(),
-		Container:      NewTopology(),
-		ContainerImage: NewTopology(),
-		Host:           NewTopology(),
-		Overlay:        NewTopology(),
+		Endpoint:       MakeTopology(),
+		Address:        MakeTopology(),
+		Process:        MakeTopology(),
+		Container:      MakeTopology(),
+		ContainerImage: MakeTopology(),
+		Host:           MakeTopology(),
+		Overlay:        MakeTopology(),
 		Sampling:       Sampling{},
 		Window:         0,
 	}
 }
 
+// Copy returns a value copy of the report.
+func (r Report) Copy() Report {
```

Comment:
r itself is a copy, but r's fields have reference semantics and need to be manually copied.